### PR TITLE
Migrate Discourse integration to the new framework

### PR DESCRIPTION
**Pull request by garrrikkotua**
# Changes proposed ✍️

### <samp>🤖 Generated by Copilot at 35081c1</samp>

Refactored the discourse integration logic to use the new service SQS queue and the integrations library. Removed the old node worker SQS queue and the `DiscourseIntegrationService` class. Added new functions and files to handle the discourse API calls, stream generation, data processing, and webhook processing. Commented out the discourse webhook endpoint temporarily.
​

### <samp>🤖 Generated by Copilot at 35081c1</samp>

> _We are the masters of discourse_
> _We integrate with force and skill_
> _We use the service queue and library_
> _We comment out the route until_

### <samp>🤖 Generated by Copilot at 35081c1</samp>

*  Refactor the webhook processing logic to use a new service SQS queue instead of the node worker SQS queue ([link](https://github.com/CrowdDotDev/crowd.dev/pull/1778/files?diff=unified&w=0#diff-e73f1bddbd02ecda55601824d80a247f1e1b26d00bf1473e5b4fff56f25943f3L6-R7), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1778/files?diff=unified&w=0#diff-e73f1bddbd02ecda55601824d80a247f1e1b26d00bf1473e5b4fff56f25943f3L88-R92), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1778/files?diff=unified&w=0#diff-c7b6581fe259349c05ed76006fd9488b4006bbb4793a1606e5ec5692d50b005dL7-R7), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1778/files?diff=unified&w=0#diff-80e1c7e63674653fe2cf5bcbe4a2212499d72014f328ad2e6b617146f0a256fdL1474-R1477))
*  Remove the discourse integration service and the integration run creation logic as they are no longer needed after the refactoring ([link](https://github.com/CrowdDotDev/crowd.dev/pull/1778/files?diff=unified&w=0#diff-5c754532401a5eebb0928f7ec08cd57f227637fc57f6a1c35c0fb3ede4611614L9), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1778/files?diff=unified&w=0#diff-5c754532401a5eebb0928f7ec08cd57f227637fc57f6a1c35c0fb3ede4611614L27), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1778/files?diff=unified&w=0#diff-80e1c7e63674653fe2cf5bcbe4a2212499d72014f328ad2e6b617146f0a256fdL1444), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1778/files?diff=unified&w=0#diff-80e1c7e63674653fe2cf5bcbe4a2212499d72014f328ad2e6b617146f0a256fdL1462-L1467))
*  Implement the integrations library for the discourse integration, which defines the type, member attributes, generate streams, process stream, process data, and post process functions for the integration ([link](https://github.com/CrowdDotDev/crowd.dev/pull/1778/files?diff=unified&w=0#diff-59a805bec16c58136d0b12aff4ad267a155ffb23083ae23e4e61f26511421be4R1-R21))
*  Implement the functions to fetch the categories, topics, posts, and user from the discourse forum using the discourse API, which are used by the integrations library to process the data for the integration ([link](https://github.com/CrowdDotDev/crowd.dev/pull/1778/files?diff=unified&w=0#diff-dcd83868458b7a50b37d66715f6e6db561f35462f9b402b01dafe19a9986e723R1-R35), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1778/files?diff=unified&w=0#diff-33a04c8ec71869f67c1771d05b5b7223057dae2f55c2856d4cf95f031366cb56R1-R62), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1778/files?diff=unified&w=0#diff-f98b86421fb7772d8255cf92251d2608c104ac6b7df858a1000362f6d1ab257cR1-R44), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1778/files?diff=unified&w=0#diff-1ebb021056d42466e6c6e9544176be0e7c9defa831b7a2f60f41a67bc950cc92R1-R40), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1778/files?diff=unified&w=0#diff-e6ef53f2fce14b7393f4e496c8634f4a4c6574d2224902840adb5130b8638a84R1-R48))
*  Implement the handler functions to generate the streams, process the streams, process the data, and process the webhook streams for the discourse integration, which are used by the integrations library to initiate and execute the integration logic ([link](https://github.com/CrowdDotDev/crowd.dev/pull/1778/files?diff=unified&w=0#diff-9bbdc6f01fcca7e3f9dea2f878991ecfd668c5b3864021968b8c7da5fffff1d3R1-R11), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1778/files?diff=unified&w=0#diff-ab24b3b2f5f8627497c61f35c9f0130d35b3b5dea6e36d12873a123c3ca79d3bR1-R168), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1778/files?diff=unified&w=0#diff-802e86fae894b1b20addceec8f98ec6ccf6a5431860a355ca1ca5152affc7a1cR1-R211), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1778/files?diff=unified&w=0#diff-a6f62144c899bc473945f618f3e0bd8632b83da75cc7e3d090d5576fb278aaaaR1-R131))

## Checklist ✅
- [x] Label appropriately with `Feature`, `Improvement`, or `Bug`.
- [ ] Add screenshots to the PR description for relevant FE changes
- [ ] New backend functionality has been unit-tested.
- [ ] API documentation has been updated (if necessary) (see [docs on API documentation](https://docs.crowd.dev/docs/updating-api-documentation)).
- [x] [Quality standards](https://github.com/CrowdDotDev/crowd-github-test-public/blob/main/CONTRIBUTING.md#quality-standards) are met.

### CODE REVIEW

**Comment by peoray** [comment]
Great work.

Quick question: When will the docs be updated so anyone can add new integrations?

cc @garrrikkotua @themarolt

**Review by epipav — COMMENTED** [review]
I added a few comments

File: services/libs/integrations/src/integrations/discourse/api/getPostsByIds.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import axios, { AxiosRequestConfig } from 'axios'
+import { RateLimitError } from '@crowd/types'
+import {
+  DiscourseConnectionParams,
+  DiscoursePostsByIdsResponse,
+  DiscoursePostsByIdsInput,
+} from '../types'
+import { IProcessStreamContext } from '../../../types'
+
+const serializeArrayToQueryString = (params: object) =>
```
[the remark this answers is not in the document]
Comment (garrrikkotua) [reply services/libs/integrations/src/integrations/discourse/api/getPostsByIds.ts:10]: done

File: services/libs/integrations/src/integrations/discourse/api/getTopics.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  try {
+    const response = await axios(config)
+    return response.data
+  } catch (err) {
+    if (err.response && err.response.status === 429) {
+      // wait 5 mins
+      throw new RateLimitError(5 * 60, 'discourse/gettopics')
+    }
+    ctx.log.error({ err, params }, 'Error while getting Discourse categories')
```
[the remark this answers is not in the document]
Comment (garrrikkotua) [reply services/libs/integrations/src/integrations/discourse/api/getTopics.ts:37]: done

File: services/libs/integrations/src/integrations/discourse/api/getUser.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import { RateLimitError } from '@crowd/types'
+import type {
+  DiscourseConnectionParams,
+  DiscourseUserResponse,
+  DisourseUserByUsernameInput,
+} from '../types'
+import { IProcessStreamContext } from '../../../types'
+
+// this methods returns ids of posts in a topic
+// then we need to parse each topic individually (can be batched)
```
[the remark this answers is not in the document]
Comment (garrrikkotua) [reply services/libs/integrations/src/integrations/discourse/api/getUser.ts:11]: done

File: services/libs/integrations/src/integrations/discourse/processStream.ts [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+        `${DiscourseStreamType.POSTS_FROM_TOPIC}:${topic.id}`,
+        {
+          topicId: topic.id,
+          topic_slug: topic.slug,
+          page: 0,
+        },
+      )
+    })
+
+    // we aslo need to trigger nextPageStream
```
[the remark this answers is not in the document]
Comment (garrrikkotua) [reply services/libs/integrations/src/integrations/discourse/processStream.ts:82]: done

**Pull request merged**